src/ajax.js

```diff
@@ -557,8 +557,14 @@ jQuery.extend( {
 			}
 		}
 
+		// Uppercase the type
+		s.type = s.type.toUpperCase();
+
+		// Determine if request has content
+		s.hasContent = !rnoContent.test( s.type );
+
 		// Convert data if not already a string
-		if ( s.data && s.processData && typeof s.data !== "string" ) {
+		if ( s.data && ( s.processData || !s.hasContent ) && typeof s.data !== "string" ) {
```

Comment:
Let me restate: if you want `jQuery.ajax` to produce a string, set `processData: true` (the default). If you want to _prevent_ `jQuery.ajax` from altering your data, set `processData: false`. Why are you setting `processData: false` when you want the `processData: true` behavior?